### ISSUE TITLE
Hooks don't run

### ISSUE THREAD

**Issue by witten** [roboll/helmfile#363]
##### Reproduction steps

I'm trying to prototype the new hooks feature, basically using the example from the README in a `helmfile.yaml`:

```yaml
releases:
- name: test
  chart: stable/chartmuseum
  hooks:
  - events: ["prepare", "cleanup"]
    command: "echo"
    args: ["{{`{{.Environment.Name}}`}}", "{{`{{.Release.Name}}`}}", "{{`{{.HelmfileCommand}}`}}\
"]
```

Then, I run the following:

```bash
$ helmfile --version
helmfile version v0.37.0
$ helmfile --namespace test sync
```

And that produces the following

```bash
Updating repo
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "incubator" chart repository
...Successfully got an update from the "energysavvy" chart repository
...Successfully got an update from the "stable" chart repository
...Successfully got an update from the "coreos" chart repository
Update Complete. ⎈ Happy Helming!⎈ 

Upgrading stable/chartmuseum
Release "test" does not exist. Installing it now.
NAME:   test
LAST DEPLOYED: Tue Sep 25 19:34:53 2018
NAMESPACE: test
STATUS: DEPLOYED

RESOURCES:
==> v1/Pod(related)
NAME                              READY  STATUS             RESTARTS  AGE
test-chartmuseum-d9584c589-6hp5h  0/1    ContainerCreating  0         0s

==> v1/Secret
NAME              TYPE    DATA  AGE
test-chartmuseum  Opaque  0     0s

==> v1/Service
NAME              TYPE       CLUSTER-IP      EXTERNAL-IP  PORT(S)   AGE
test-chartmuseum  ClusterIP  100.65.193.150  <none>       8080/TCP  0s

==> v1beta1/Deployment
NAME              DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
test-chartmuseum  1        1        1           0          0s


NOTES:
** Please be patient while the chart is being deployed **

Get the ChartMuseum URL by running:

  export POD_NAME=$(kubectl get pods --namespace test -l "app=chartmuseum" -l "release=test" -o jsonpath="{.items[0].metadata.name}")
  echo http://127.0.0.1:8080/
  kubectl port-forward $POD_NAME 8080:8080 --namespace test


helmfile.yaml: basePath=/app/tmp
```

##### Actual behavior

The `echo`s from the hooks do not appear in the output. Additionally, if I change the `echo` to an invalid command, those do not error or appear to run at all.

##### Expected behavior

Hooks run and echo output to stdout/stderr.

**Comment by mumoshu** [comment]
@witten Thanks for the report!

> The echos from the hooks do not appear in the output.

Does it appear once you added `--log-level debug` like `helmfile --log-level debug apply`?
I thought I suppressed the logs by default not to bloat the screen, but worth reconsidering that.

> if I change the echo to an invalid command, those do not error or appear to run at all.

This seems like a bug. I'll fix it asap.

**Comment by mumoshu** [comment]
@witten Hey! Please try `helmfile --log-level debug COMMAND` with v0.37.0 that includes #364

I'm still wondering if we really should print stdout emitted from hook commands. Thoughts?

**Comment by witten** [comment]
Yup, sorry for the delay. Here it is:

Still using the released v0.37.0:

```bash
helmfile --log-level debug --namespace test sync 
Processing helmfile.yaml
second-pass rendering result of "helmfile.yaml":
 0: releases:
 1: - name: test 
 2:   chart: stable/chartmuseum
 3:   hooks:
 4:   - events: ["prepare", "cleanup"]
 5:     command: "echo"
 6:     args: ["{{.Environment.Name}}", "{{.Release.Name}}", "{{.HelmfileCommand}}\
 7: "]
 8: 

Updating repo
exec: helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "incubator" chart repository
...Successfully got an update from the "energysavvy" chart repository
...Successfully got an update from the "stable" chart repository
...Successfully got an update from the "coreos" chart repository
Update Complete. ⎈ Happy Helming!⎈ 

Upgrading stable/chartmuseum
exec: helm upgrade --install --reset-values test stable/chartmuseum --namespace test
Release "test" has been upgraded. Happy Helming!
LAST DEPLOYED: Wed Sep 26 17:13:47 2018
NAMESPACE: test
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/Deployment
NAME              DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
test-chartmuseum  1        1        1           1          3m

==> v1/Pod(related)
NAME                              READY  STATUS   RESTARTS  AGE
test-chartmuseum-d9584c589-bbb9z  1/1    Running  0         3m

==> v1/Secret
NAME              TYPE    DATA  AGE
test-chartmuseum  Opaque  0     3m

==> v1/Service
NAME              TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)   AGE
test-chartmuseum  ClusterIP  100.67.80.112  <none>       8080/TCP  3m


NOTES:
** Please be patient while the chart is being deployed **

Get the ChartMuseum URL by running:

  export POD_NAME=$(kubectl get pods --namespace test -l "app=chartmuseum" -l "release=test" -o jsonpath="{.items[0].metadata.name}")
  echo http://127.0.0.1:8080/
  kubectl port-forward $POD_NAME 8080:8080 --namespace test


helmfile.yaml: basePath=/app/tmp
hook[echo]: triggered by event "cleanup"
```

**Comment by witten** [comment]
Here's the same thing with Helmfile master. Looks closer to your output, which indicates it's probably fixed with your recent changes. However, I don't see the actual `echo` output on the cleanup, even though it appears to run:

```
$ ./helmfile.master --log-level debug --namespace test sync 
Processing helmfile.yaml
second-pass rendering result of "helmfile.yaml":
 0: releases:
 1: - name: test 
 2:   chart: stable/chartmuseum
 3:   hooks:
 4:   - events: ["prepare", "cleanup"]
 5:     command: "echo"
 6:     args: ["{{.Environment.Name}}", "{{.Release.Name}}", "{{.HelmfileCommand}}\
 7: "]
 8: 

helmfile.yaml: basePath=/app/tmp
hook[echo]: triggered by event "prepare"

hook[echo]: default test sync


Upgrading stable/chartmuseum
exec: helm upgrade --install --reset-values test stable/chartmuseum --namespace test
Release "test" has been upgraded. Happy Helming!
LAST DEPLOYED: Wed Sep 26 17:25:36 2018
NAMESPACE: test
STATUS: DEPLOYED

RESOURCES:
==> v1/Secret
NAME              TYPE    DATA  AGE
test-chartmuseum  Opaque  0     15m

==> v1/Service
NAME              TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)   AGE
test-chartmuseum  ClusterIP  100.67.80.112  <none>       8080/TCP  15m

==> v1beta1/Deployment
NAME              DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
test-chartmuseum  1        1        1           1          15m

==> v1/Pod(related)
NAME                              READY  STATUS   RESTARTS  AGE
test-chartmuseum-d9584c589-bbb9z  1/1    Running  0         15m


NOTES:
** Please be patient while the chart is being deployed **

Get the ChartMuseum URL by running:

  export POD_NAME=$(kubectl get pods --namespace test -l "app=chartmuseum" -l "release=test" -o jsonpath="{.items[0].metadata.name}")
  echo http://127.0.0.1:8080/
  kubectl port-forward $POD_NAME 8080:8080 --namespace test


helmfile.yaml: basePath=/app/tmp
hook[echo]: triggered by event "cleanup"
```

**Comment by mumoshu** [comment]
@witten Thanks for your patience! #367 should fix it.

**Comment by mumoshu** [comment]
And v0.38.3 has been shipped with #367.

**Comment by witten** [comment]
Awesome, thank you for the quick turn-around!
